### PR TITLE
Add support for localnet

### PR DESCRIPTION
**Pull request by mfornet**
Solve #13

### CODE REVIEW

File: nearuplib/localnet.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    rmtree('logs', ignore_errors=True)
+    mkdir('logs')
+
+    # Spawn network
+    pid_fd = open('node.pid', 'w')
+    for i in range(0, args.num_nodes):
+        proc = run_binary(args.binary_path, join(args.home, f'node{i}'), 'run',
+                          boot_nodes=f'{pk}@127.0.0.1:24567' if i > 0 else None, output=f'logs/node{i}')
+        print(proc.pid, file=pid_fd)
+    pid_fd.close()
```
Comment (ailisp) [inline nearuplib/localnet.py:77]: This is clever! So if i understand correctly, nearup will exit after several run_binary. The nodes are still running in current shell? Until you localnet --stop it?

**Review by bowenwang1996 — COMMENTED** [review]
why do you need a separate file for starting localnet? Input should be handled in main.py

File: nearuplib/localnet.py (hunk 1 above)
Comment (mfornet) [reply nearuplib/localnet.py:77]: Yes, that's correct

File: nearuplib/localnet.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    args.binary_path = join(args.binary_path, 'near')
+
+    path = 'node.pid'
+    if args.stop:
+        if exists(path):
+            with open(path) as f:
+                for x in f.readlines():
+                    pid = int(x.strip(' \n'))
+                    print("Killing:", pid)
+                    kill(pid, SIGTERM)
```
Comment (ailisp) [inline nearuplib/localnet.py]: There's a chance node killed before and pid refer to incorrect running processes. I suggest do a quick check if it name has near, skip if it doesn't

File: nearuplib/localnet.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            unlink(path)
+    else:
+        if exists(path):
+            print("There is already a test running. Stop it using:")
+            print("python3 start_localnet.py -k")
```
Comment (ailisp) [inline nearuplib/localnet.py]: This need to be updated to nearup too

**Review by ailisp — APPROVED** [review]

**Comment by mfornet** [comment]
> why do you need a separate file for starting localnet? Input should be handled in main.py

@bowenwang1996  The main reason is because I just moved an existent script from [localnet-script repository](https://github.com/nearprotocol/localnet-script) to here.

I tried to reuse the same parser, but the arguments are different, as it requires also number of nodes/shards. And more importantly have a different way to stop all nodes.

File: nearuplib/localnet.py (hunk 2 above)
Comment (mfornet) [reply nearuplib/localnet.py]: @ailisp How can this be done in such a way that it supports multiple platforms? One alternative I found is to use [psutil](https://github.com/giampaolo/psutil) but I think it is a bad idea having a dependency in nearup.

**Pull request merged**